File: 백준/Silver/10709. 기상캐스터/기상캐스터.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        StringBuilder sb = new StringBuilder();

        int h = Integer.parseInt(st.nextToken());
        int w = Integer.parseInt(st.nextToken());

        char[][] cloud = new char[h][w];
        int[][] result = new int[h][w];

        for (int i = 0; i < h; i++) {
            cloud[i] = br.readLine().toCharArray();
            Arrays.fill(result[i], -1);
        }

        for (int i = 0; i < h; i++) {
            for (int j = 0; j < w; j++) {
                if (cloud[i][j] == 'c') {
                    result[i][j] = 0;
                } else if (j > 0 && result[i][j - 1] >= 0) {
                    result[i][j] = result[i][j - 1] + 1;
                }
            }
        }

        for (int i = 0; i < h; i++) {
            for (int j = 0; j < w; j++) {
                sb.append(result[i][j]).append(" ");
            }
            sb.append("\n");
        }

        System.out.println(sb);
    }
}

File: 백준/Silver/10709. 기상캐스터/기상캐스터.py
h, w = map(int, input().split())
cloud = [list(input()) for _ in range(h)]
result = [[-1] * w for _ in range(h)]

for i in range(h):
    for j in range(w):
        if cloud[i][j] == 'c':
            result[i][j] = 0
        elif j > 0 and result[i][j - 1] >= 0:
            result[i][j] = result[i][j - 1] + 1

for i in range(h):
    print(*result[i])